introduction-to-datasets.md

```diff
@@ -0,0 +1,142 @@
+---
+layout: page
+title: Introduction to Datasets
+---
+
+A dataset is a collection of records, similar to a relational database table. Records are similar to table rows, but the columns can contain not only strings or numbers, but also nested data structures such as lists, maps, and other records.
```

Comment:
Here's a version that solves both problems. It uses _like_ and uses a single _but_ to clarify how they are not identical to rows: "Records are like table rows, but can contain lists, maps, and nested records, in addition to primitive types such as strings or numbers."